src/framework/ui/modal/modal.component.tsx

```diff
@@ -111,154 +94,100 @@ export type ModalProps = ViewProps & ComponentProps;
-    return;
+    return (
+      <View
+        {...restProps}
+        {...measuringProps}
+        key={TAG_CHILD}
+        style={[styles.container, absoluteRelatedStyle, style]}>
+        {children}
+      </View>
+    );
   };
 
-  private onStartShouldSetResponderCapture = (): boolean => false;
+  private renderModal = (): React.ReactElement => {
+    const { backdropStyle } = this.props;
+    const modal: React.ReactElement<ViewProps> = this.renderBaseModal();
+
+    return backdropStyle ? (
+      <React.Fragment>
+        <View
+          pointerEvents='box-none'
+          style={[styles.backdropBaseStyles, backdropStyle]}/>
+        {modal}
+      </React.Fragment>
+    ) : modal;
+  };
 
-  private renderComponentChild = (source: React.ReactElement<any>): React.ReactElement<any> => {
-    return React.cloneElement(source, {
-      ...source.props,
-      onCloseModal: this.closeModal,
-      style: {
-        ...source.props.style,
-        ...StyleSheet.flatten(this.props.style),
-      },
+  private renderMeasureNode = (): React.ReactElement<MeasureNodeProps> => {
+    const modal: React.ReactElement = this.renderBaseModal();
+    const measureStyledModal: React.ReactElement = React.cloneElement(modal, {
+      style: [modal.props.style, styles.hiddenModal],
+      key: TAG_CHILD,
     });
-  };
 
-  private renderComponentChildren = (source: React.ReactNode): React.ReactElement<any>[] => {
-    return React.Children.map(source, this.renderComponentChild);
+    return (
+      <MeasureNode onResult={this.onMeasure}>
+        {[measureStyledModal]}
+      </MeasureNode>
+    );
   };
 
-  private renderWithBackDrop = (component: React.ReactElement<ViewProps>) => (
-    <TouchableOpacity
-      style={styles.container}
-      onPress={this.closeOnBackdrop}
-      activeOpacity={1}>
-      {component}
-    </TouchableOpacity>
-  );
-
-  private renderWithoutBackDrop = (component: React.ReactElement<ViewProps>) => (
-    <View style={styles.notVisibleWrapper}>
-      <View
-        style={styles.container}
-        pointerEvents='none'/>
-      {component}
-    </View>
-  );
-
-  private renderComponent = (): React.ReactElement<TouchableOpacityProps | ViewProps> => {
-    const { animationType, children, isBackDropAllowed, ...derivedProps } = this.props;
-    const animationStyle: StyleType = this.getAnimationStyle(animationType);
-    const componentChildren: React.ReactElement<any>[] = this.renderComponentChildren(children);
-
-    const dialog: React.ReactElement<ViewProps> =
-      <Animated.View
-        {...derivedProps}
-        style={[animationStyle, styles.animatedWrapper]}
-        onStartShouldSetResponder={this.onStartShouldSetResponder}
-        onResponderRelease={this.onResponderRelease}
-        onStartShouldSetResponderCapture={this.onStartShouldSetResponderCapture}
-        pointerEvents='box-none'>
-        {componentChildren}
-      </Animated.View>;
+  private onMeasure = (result: MeasureResult): void => {
```

Comment:
should be declared above all renders

Reply:
Got it, will do. But anyway what's the reason?